prow/config/inrepoconfig.go

```diff
@@ -129,8 +129,12 @@ func prowYAMLGetter(
 
 func ensureHeadCommits(repo git.RepoClient, headSHAs ...string) error {
 	for _, sha := range headSHAs {
-		if err := repo.Fetch(sha); err != nil {
-			return fmt.Errorf("failed to fetch headSHA: %s: %v", sha, err)
+		// This is best effort. No need to check for error
+		exists, _ := repo.CommitExists(sha)
+		if !exists {
```

Comment:
one line please to reduce the scope of `exists`

Reply:
Done